### PR TITLE
chore(dependencies): update nextjs to version 14

### LINKED ISSUE #333: [FEATURE] Update the packages in the repository
### Why do we need this improvement?

This should help us get rid of the security vulnarablities of the current `Next.js 12` and upgrade it to `Next 14`

### How will this change help?

It will remove  the version incomaptablities that may arise in the future.

### Screenshots

Screenshot from 2024-06-11 17-03-33

### How could it be implemented/designed?

__Resource__:-  https://nvd.nist.gov/vuln/detail/CVE-2023-46298

1. We need to upgrade the versions in package.json 
2. Check if all the features/components in the site are still functional
3. Provide resources utilised in the pull request

### 🚧 Breaking changes

Yes

### 👀 Have you checked for similar open issues?

- [X] I checked and didn't find a similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/asyncapi/.github/blob/master/CONTRIBUTING.md)

### Are you willing to work on this issue?

No, someone else can work on it.

### PR DESCRIPTION
**Pull request by jerensl**
**Description**
This upgrade required changing the nvmrc to use nvm 18>, because NextJS 14 not supporting node below version 18.
I'm also updating ReactJS and ReactDOM to the latest version to support NextJS 14

**Related issue(s)**
Resolves #333

### CODE REVIEW

**Comment by ashmit-coder** [comment]
Hey @jerensl the changes look good to me. Have to tested for breaking in the website if any?

**Comment by jerensl** [comment]
> Hey @jerensl the changes look good to me. Have to tested for breaking in the website if any?

I checked the staging preview and it looks fine to me, if you find any bug feel free to ping me again

**Comment by ashmit-coder** [comment]
Hey @jerensl can you mention any resources you followed. 

The site does not seem to have bugs either

File: package.json [hunk 1]
```diff
@@ -5,9 +5,9 @@
   "scripts": {
     "dev": "next dev",
     "build": "next build",
-    "start": "next start",
+    "start": "npx serve ./out/",
```
Comment (ashmit-coder) [inline package.json:8]: Why wont `next start` work over here?

**Comment by jerensl** [comment]
> ### ✅ Deploy Preview for _peaceful-ramanujan-288045_ ready!
> Name	Link
> 🔨 Latest commit	[14757c8](https://github.com/asyncapi/conference-website/commit/14757c8e020bb3ed38226d457d146f09dc94e56f)
> 🔍 Latest deploy log	https://app.netlify.com/sites/peaceful-ramanujan-288045/deploys/666856db1eb97200089af04c
> 😎 Deploy Preview	https://deploy-preview-335--peaceful-ramanujan-288045.netlify.app
> 📱 Preview on mobile	
> Toggle QR Code...
> _To edit notification comments on pull requests, go to your [Netlify site configuration](https://app.netlify.com/sites/peaceful-ramanujan-288045/configuration/deploys#deploy-notifications)._

@ashmit-coder I mean this one, if I'm not mistaken this is staging website generated by netlify right?

File: package.json (hunk 1 above)
Comment (ashmit-coder) [reply package.json:8]: In my opinion let's just have next start cause the website is not entirely static / server side rendered

Comment (jerensl) [reply package.json]: After did the changes I will rollback to use export again after it breaks the staging build, looks like Netlify has their own command as referred to the image below
Screenshot 2024-06-12 172329

Comment (ashmit-coder) [reply package.json:8]: I see let's get @AceTheCreator opinion on this. Cause I don't have access to netlify

Comment (ashmit-coder) [reply package.json:8]: Hey @jerensl 

have a look at the Netlify.toml, the Node version is not as said and you can change the build and deploy command from there I guess.

So look into it :)

**Comment by AceTheCreator** [comment]
@ashmit-coder @jerensl I don't really understand y'all argument. Do you mind making it more understandable?

**Comment by jerensl** [comment]
> > @ashmit-coder @jerensl I don't really understand y'all argument. Do you mind making it more understandable?
> 
> I dont think we had proper communication over it, I will talk to @jerensl for now and let you know soon!

Let me clarify, as I understand there are 2 ways to deploy a web app:
1. Static file(HTML, CSS, and Javascript), mostly it being stored in the storage and replicated by a CDN across the globe. In here what matters is the size and bandwidth most of the services offer 100GB of bandwidth for free, which is why this option is cheaper than the other one. In NextJS the way it did by using static export
2. The server side(pay /request), mostly requires a NodeJS server and the content is dynamic. In Next JS we did it by running next/cli start

Most of the CDN services like Netlify etc, it will be running the `npm build` for the build from `package.json` scripts but not in the case after the build or serving the web app, most of them have their own default command, and mostly not same as `package.json` script `npm run install`

That's why I and @ashmit-coder have a different opinion on `package.json` `npm run start` script command. He wanted to use `next/cli start` and I suggested `npx serve ./out/` & export build

**Comment by jerensl** [comment]
After making changes as requested, I decided not to continue this work anymore because don't have the bandwidth to deal with that now thanks. Anyone can take this issue from now

**Pull request closed without merging**